File: crm-frontend/src/services/leadService.js
import httpClient from '../api/httpClient'

export const leadService = {
  async getAll(page = 0, size = 10) {
    const response = await httpClient.get('/leads', {
      params: { page, size },
    })
    return response.data
  },

  async getById(id) {
    const response = await httpClient.get(`/leads/${id}`)
    return response.data
  },

  async create(leadData) {
    const response = await httpClient.post('/leads', leadData)
    return response.data
  },

  async update(id, leadData) {
    const response = await httpClient.put(`/leads/${id}`, leadData)
    return response.data
  },

  async delete(id) {
    const response = await httpClient.delete(`/leads/${id}`)
    return response.data
  },

  async searchByName(name, page = 0, size = 10) {
    const response = await httpClient.get('/leads/search/name', {
      params: { name, page, size },
    })
    return response.data
  },

  async searchByEmail(email, page = 0, size = 10) {
    const response = await httpClient.get('/leads/search/email', {
      params: { email, page, size },
    })
    return response.data
  },

  async searchByCompany(company, page = 0, size = 10) {
    const response = await httpClient.get('/leads/search/company', {
      params: { company, page, size },
    })
    return response.data
  },

  async assignToUser(leadId, userId) {
    const response = await httpClient.put(`/leads/${leadId}/assign/${userId}`)
    return response.data
  },
}

File: src/main/java/com/crm/service/LeadService.java
package com.crm.service;

import com.crm.dto.request.LeadCreateRequest;
import com.crm.dto.request.LeadUpdateRequest;
import com.crm.dto.response.LeadResponse;
import com.crm.util.PagedResponse;
import org.springframework.data.domain.Pageable;

public interface LeadService {
    LeadResponse create(LeadCreateRequest request);

    LeadResponse update(Long id, LeadUpdateRequest request);

    LeadResponse getById(Long id);

    PagedResponse<LeadResponse> list(Pageable pageable);

    void delete(Long id);

    PagedResponse<LeadResponse> searchByName(String name, Pageable pageable);

    PagedResponse<LeadResponse> searchByEmail(String email, Pageable pageable);

    PagedResponse<LeadResponse> searchByCompany(String company, Pageable pageable);

    LeadResponse assignToUser(Long leadId, Long userId);
}
